### ISSUE TITLE
"unexpected token" macro-error on derive macro

### ISSUE THREAD

**Issue by GREsau** [rust-lang/rust-analyzer#17968]
**rust-analyzer version**: `0.3.2086-standalone (7106cd3be 2024-08-25)`

**rustc version**: `rustc 1.80.1 (3f5fd8dd4 2024-08-06)`

**editor or extension**: VSCode, rust-analyzer v0.3.2086 (error does not occur with v0.3.2078)

**relevant settings**: n/a

**repository link (if public, optional)**: n/a

**code snippet to reproduce**: 
`cargo add schemars` (reproducible both with stable v0.8.21 and prerelease v1.0.0-alpha.11)
```rust
#[derive(schemars::JsonSchema)]
pub struct X;

fn main() {
    println!("{:?}", schemars::schema_for!(X))
}
```

![image](https://github.com/user-attachments/assets/01dc135f-20ce-41ef-a8e2-3fbd7556c1eb)

The code compiles and runs fine. When running `cargo expand` and using schemars v1.0.0-alpha.11, the derive expands to:

```rust
const _: () = {
    #[automatically_derived]
    #[allow(unused_braces)]
    impl schemars::JsonSchema for X {
        fn schema_name() -> schemars::_alloc::borrow::Cow<'static, str> {
            schemars::_alloc::borrow::Cow::Borrowed("X")
        }
        fn schema_id() -> schemars::_alloc::borrow::Cow<'static, str> {
            schemars::_alloc::borrow::Cow::Borrowed("rust_test::X")
        }
        fn json_schema(generator: &mut schemars::SchemaGenerator) -> schemars::Schema {
            generator.subschema_for::<()>()
        }
    }
};
```
...which rust-analyzer has no problems with.

**Comment by rever96** [comment]
I got the same problem.
Go back to v0.3.2078 until they will fix the problem

**Comment by SenojLuap** [comment]
Not sure if it's the same issue, but I get a similar looking error while using rocket's `get` macro
![image](https://github.com/user-attachments/assets/0299d50b-0954-4f3a-a8bc-4df84ff2e8ff)

**Comment by ChayimFriedman2** [comment]
Oh, this is already fixed on latest rust-analyzer.

@rustbot release-assignment

**Comment by kerminek** [comment]
> Not sure if it's the same issue, but I get a similar looking error while using rocket's `get` macro ![image](https://private-user-images.githubusercontent.com/1019888/361451831-0299d50b-0954-4f3a-a8bc-4df84ff2e8ff.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ3MTMyODUsIm5iZiI6MTcyNDcxMjk4NSwicGF0aCI6Ii8xMDE5ODg4LzM2MTQ1MTgzMS0wMjk5ZDUwYi0wOTU0LTRmM2EtYThiYy00ZGY4NGZmMmU4ZmYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MjZUMjI1NjI1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGZhZTNkZDkwMTU5YmEwZWIxNzM5OTM3Njg4YWRiOGQ5NGExNTk1ODYxNzkyZTJiYzFmMjdiZGU3NWY3ZDliYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.xfj9U7Q8U-OiHu_6NA76l6gTmbRd4YGPFmROIw_WKK4)

Same
![Zrzut ekranu 2024-08-27 o 00 57 25](https://github.com/user-attachments/assets/8b47302a-4b4e-43b2-ae1e-91afe28c7c1a)

**Comment by ChayimFriedman2** [comment]
@kerminek @SenojLuap The problem is `module_path!()` not expanding correctly. If `rocket`'s `#[get]` uses it, it is likely the same problem indeed (you can check by r-a "Expand macro recursively"), do not use cargo-expand because it will expand `module_path!()` too.

**Comment by Spycemyster** [comment]
> Not sure if it's the same issue, but I get a similar looking error while using rocket's `get` macro ![image](https://private-user-images.githubusercontent.com/1019888/361451831-0299d50b-0954-4f3a-a8bc-4df84ff2e8ff.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ3MzA4NjksIm5iZiI6MTcyNDczMDU2OSwicGF0aCI6Ii8xMDE5ODg4LzM2MTQ1MTgzMS0wMjk5ZDUwYi0wOTU0LTRmM2EtYThiYy00ZGY4NGZmMmU4ZmYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MjdUMDM0OTI5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzI3YWE3Yjk3ZTZiYzQ2Y2U3NzlhNWMzYTY2ZGE1NmMwOTM3ZGFmNzMxZmMyYmE3ZmZkNjJiY2E0Y2FlODgyZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.dPyHY16Sk4MjwhcvaOgzkBI1kCYKaB7mja-KZMeuRig)

Switching to rust-analyzer 0.3.2078 fixes it for now (credit to @rever96)

**Comment by lnicola** [comment]
v0.3.2089 should have a fix for this.

**Comment by GREsau** [comment]
Thanks, I can confirm this is now resolved after updating the extension to v0.3.2089 🙂

**Comment by SenojLuap** [comment]
Same. Thanks!
